khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:data/api/ball_score/ball_score_model.dart';
+import 'package:data/api/innings/inning_model.dart';
+import 'package:data/api/match/match_model.dart';
+import 'package:data/service/ball_score/ball_score_service.dart';
+import 'package:data/service/innings/inning_service.dart';
+import 'package:data/service/match/match_service.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:khelo/domain/extensions/context_extensions.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_commentary_view.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_highlight_view.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_info_view.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_overs_view.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_scorecard_view.dart';
+import 'package:khelo/ui/flow/matches/match_detail/components/match_detail_squad_view.dart';
+
+part 'match_detail_tab_view_model.freezed.dart';
+
+final matchDetailTabStateProvider = StateNotifierProvider.autoDispose<
+    MatchDetailTabViewNotifier, MatchDetailTabState>(
+  (ref) => MatchDetailTabViewNotifier(
+    ref.read(matchServiceProvider),
+    ref.read(inningServiceProvider),
+    ref.read(ballScoreServiceProvider),
+  ),
+);
+
+class MatchDetailTabViewNotifier extends StateNotifier<MatchDetailTabState> {
+  final MatchService _matchService;
+  final InningsService _inningService;
+  final BallScoreService _ballScoreService;
+  late String _matchId;
+
+  MatchDetailTabViewNotifier(
+    this._matchService,
+    this._inningService,
+    this._ballScoreService,
+  ) : super(const MatchDetailTabState());
+
+  void setData(String matchId) {
+    _matchId = matchId;
+    loadMatch();
+  }
+
+  void loadMatch() {
+    state = state.copyWith(loading: true);
+    _matchService.getMatchStreamById(_matchId).listen(
+      (match) {
+        state = state.copyWith(
+            match: match,
+            highlightTeamId: match.teams.first.team.id,
+            error: null);
+        if (!state.inningsQueryListenerSet) {
+          loadInnings();
+        }
+      },
+      onError: (e, stack) {
+        state = state.copyWith(error: e, loading: false);
+        debugPrint(
+            "MatchDetailTabViewNotifier: error while load match -> $e. \n stack -> $stack");
+      },
+    );
+  }
```

Comment:
I think it is good practice to add the `StreamSubscription` variable and dispose of it on the Riverpod dispose method